llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -9790,10 +9792,11 @@ BoUpSLP::getEntryCost(const TreeEntry *E, ArrayRef<Value *> VectorizedVals,
   case Instruction::AShr:
   case Instruction::And:
   case Instruction::Or:
-  case Instruction::Xor: {
+  case Instruction::Xor:
+  case Instruction::Freeze: {
```

Comment:
I rather doubt it should be here. I think the cost of freeze must be 0 always

Reply:
Thanks for the feedback. You are correct. I will get that fixed.